model/src/test/java/com/epam/esm/dao/impl/GiftCertificateDaoImplTest.java

```diff
@@ -0,0 +1,174 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.config.H2Config;
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.entity.filter.SearchFilter;
+import com.epam.esm.exception.DaoException;
+import com.epam.esm.exception.ErrorCodes;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.function.Executable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = H2Config.class)
+@ActiveProfiles("test")
+@Transactional
+class GiftCertificateDaoImplTest {
+
+    @Autowired
+    GiftCertificateDaoImpl giftCertificateDao;
+
+    @Autowired
+    TagDao tagDao;
+
+    GiftCertificate giftCertificate1;
+    GiftCertificate giftCertificate2;
+    List<GiftCertificate> giftCertificateList;
+
+    @BeforeEach
+    void setUp() {
+        Tag tag1 = new Tag(1, "tag1");
+        Tag tag2 = new Tag(2, "tag2");
+        Tag tag3 = new Tag(3, "tag3");
+        giftCertificate1 = GiftCertificate.builder()
+                .id(1L).name("gift1").description("description")
+                .tags(List.of(tag1, tag2, tag3))
+                .price(BigDecimal.valueOf(99.99)).duration(30)
+                .createDate(LocalDateTime.parse("2023-02-17T10:48:40.303950"))
+                .lastUpdateDate(LocalDateTime.parse("2023-02-17T10:48:40.303950"))
+                .build();
+        giftCertificate2 = GiftCertificate.builder()
+                .id(2L).name("gift2").description("description2")
+                .tags(List.of(tag2, tag3))
+                .price(BigDecimal.valueOf(22.22)).duration(90)
+                .createDate(LocalDateTime.parse("2023-02-17T10:48:40.303950"))
+                .lastUpdateDate(LocalDateTime.parse("2023-02-17T10:48:40.303950"))
+                .build();
+        giftCertificateList = List.of(giftCertificate1, giftCertificate2);
+    }
+
+    @Test
+    void create_ShouldReturnCreatedCertificate() {
+        GiftCertificate certificate = new GiftCertificate();
```

Comment:
Can you move such structures to some ModelFactory utility class that will contain all factory methods for such entities creation